data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/OperatorConfigurationTest.java

```diff
@@ -0,0 +1,57 @@
+package org.opensearch.dataprepper.expression;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.dataprepper.expression.util.TestObject;
+
+import java.util.stream.Stream;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.isA;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+class OperatorConfigurationTest {
+    @InjectMocks
+    private OperatorConfiguration configuration;
+
+    @ParameterizedTest
+    @MethodSource("validTestArgProvider")
+    void equalOperator(final Object lhs, final Object rhs, final boolean isArgsEqual) {
+        final GenericEqualOperator genericEqualOperator = configuration.equalOperator();
+        assertThat(genericEqualOperator, isA(Operator.class));
+        assertThat(
+                lhs + " == " + rhs + " should equal is " + isArgsEqual,
+                genericEqualOperator.evaluate(lhs, rhs),
+                is(isArgsEqual)
+        );
+    }
+
+    private static Stream<Arguments> validTestArgProvider() {
+        return Stream.of(
+                Arguments.of(10, 10, true),
```

Comment:
We should have some negative test cases for some of these same combinations.

e.g. `Arguments.of(10, 9, false)`.